case_conversion/case_parse.py

```diff
@@ -85,167 +209,73 @@ def parse_case(var, detect_acronyms=True, acronyms=[], preserve_case=False):
-                    while True:
-                        m = rac.search(acstr, n)
-                        if not m:
-                            break
-
-                        a, b = m.start(), m.end()
-                        n = b
-
-                        # Make sure found acronym doesn't overlap with others.
-                        ok = True
-                        for r in range_list:
-                            if a < r[1] and b > r[0]:
-                                ok = False
-                                break
-
-                        if ok:
-                            range_list.append((a, b))
-                            for j in xrange(a, b):
-                                not_range.remove(j)
-
-                # Add remaining letters as ranges.
-                for nr in not_range:
-                    range_list.append((nr, nr+1))
-
-                # No ranges will overlap, so it's safe to sort by lower bound,
-                # which sort() will do by default.
-                range_list.sort()
-
-                # Remove original letters in word list.
-                for j in xrange(s, i):
-                    del words[s]
-
-                # Replace them with new word grouping.
-                for j in xrange(len(range_list)):
-                    r = range_list[j]
-                    words.insert(s+j, acstr[r[0]:r[1]])
-
-                return s+len(range_list)-1
-        else:
-            # Fallback to simple acronym detection.
-            def check_acronym(s, i):
-                # Combine each letter into a single string.
-                acronym = ''.join(words[s:i])
+    return words, separator, was_upper
+
+
+def parse_case(string, acronyms=None, preserve_case=False):
+    """
+    Parse a stringiable into a list of words.
+
+    Also returns the case type, which can be one of the following:
+        - upper: All words are upper-case.
+        - lower: All words are lower-case.
+        - pascal: All words are title-case or upper-case. Note that the
+                  stringiable may still have separators.
+        - camel: First word is lower-case, the rest are title-case or
+                 upper-case. stringiable may still have separators.
+        - mixed: Any other mixing of word casing. Never occurs if there are
+                 no separators.
+        - unknown: stringiable contains no words.
+    Also returns the first separator character, or False if there isn't one.
 
-                # Remove original letters in word list.
-                for j in xrange(s, i):
-                    del words[s]
+    """
 
-                # Replace them with new word grouping.
-                words.insert(s, ''.join(acronym))
+    words, separator, was_upper = _separate_words(string)
 
-                return s
+    if acronyms:
+        # Use advanced acronym detection with list
+        acronyms = _sanitize_acronyms(acronyms)
+        check_acronym = _advanced_acronym_detection
+    else:
```

Comment:
That comment about falling back to simple acronym detection can be moved here.

Reply:
Actually, we are now going to always perform acronym detection. Maybe it indeed makes sense to be able to disable it. What about allowing to set the parameters `acronyms` to `False`? In which case no acronym detection would take place?

Reply:
We need to decide what the default behavior should be. If we want default behavior to be that acronyms are not detected, then check if `acronyms is None` in order to skip. If we want opposite, then check for `False`. Personally I think acronyms should be by default off.